scripts/generate_image.ps1

```diff
@@ -0,0 +1,175 @@
+echo "Installing Go..."
+Start-Process "msiexec.exe" -ArgumentList "/quiet /i ${env:TEMP}\go-installer.msi" -Wait
+New-Item -ItemType directory -Path "${WORK_ROOT}\go\bin"
+New-Item -ItemType directory -Path "${WORK_ROOT}\go\src"
+[Environment]::SetEnvironmentVariable("Path", $env:Path + ";C:\Go\bin;${WORK_ROOT}\go\bin", [EnvironmentVariableTarget]::Machine)
+[Environment]::SetEnvironmentVariable("GOPATH", "${WORK_ROOT}\go", [EnvironmentVariableTarget]::Machine)
+$env:Path=$env:Path + ";C:\Go\bin;${WORK_ROOT}\go\bin"
+$env:GOPATH="${WORK_ROOT}\go"
+
+echo "Checking out ${GIT_REPO} at ${COMMIT_SHA}"
+git clone "${GIT_REPO}" "${BUILD_ROOT}" 2>&1 | % {$_.ToString()}
+cd "${BUILD_ROOT}"
+git config --add remote.origin.fetch '+refs/pull/*/head:refs/remotes/origin/pr/*'
+git fetch 2>&1 | % {$_.ToString()}
+git checkout "${COMMIT_SHA}" 2>&1 | % {$_.ToString()}
+
+echo "Starting Docker..."
+Start-Service -Name 'docker'
+
+echo "Waiting for Docker to start..."
+$dockerSrv = Get-Service -Name 'docker'
+$stat = $dockerSrv.WaitForStatus('Running', '00:15:00')
+
+echo "Waiting for Docker to become responsive..."
+docker ps
+if (${LastExitCode} -ne 0) {
+  echo "Docker ps didn't go well." -logLevel "ERROR"
+  echo $_.Exception.Message -logLevel "ERROR"
+}
+
+echo "-------------------------------------------------------------------------------"
+echo "Generating Windows Images..."
+echo "-------------------------------------------------------------------------------"
+
+$ErrorActionPreference="Stop"
+
+$application = Get-WmiObject Win32_Product -filter "Name='AWS Tools for Windows'"
+if ($application) {
+    $application.Uninstall()
+}
+Install-Module -Name AWSPowerShell -Force
+Import-Module AWSPowerShell
+
+echo "Pulling microsoft/windowsservercore:latest..."
+docker pull microsoft/windowsservercore:latest
+
+$env:ECS_WINDOWS_TEST_DIR="${BUILD_ROOT}"
+$env:ECS_FTEST_TMP="${WORK_ROOT}\ftest_temp"
+New-Item -Path "${env:ECS_FTEST_TMP}" -Type directory
+
+echo "${BUILD_ROOT}"
+
+# Prepare dependencies
+Invoke-Expression "${BUILD_ROOT}\misc\windows-iam\Setup_Iam_Images.ps1"
+Invoke-Expression "${BUILD_ROOT}\misc\volumes-test\build.ps1"
+Invoke-Expression "${BUILD_ROOT}\misc\image-cleanup-test-images\build.ps1"
+Invoke-Expression "${BUILD_ROOT}\misc\stats-windows\build.ps1"
+Invoke-Expression "${BUILD_ROOT}\misc\netkitten\build.ps1"
+Invoke-Expression "${BUILD_ROOT}\misc\windows-listen80\Setup_Listen80.ps1"
+Invoke-Expression "${BUILD_ROOT}\misc\windows-telemetry\build.ps1"
+Invoke-Expression "${BUILD_ROOT}\misc\windows-python\build.ps1"
+Invoke-Expression "${BUILD_ROOT}\misc\container-health-windows\build.ps1"
+
+$sAccessKey=(Get-SSMParameter -Name SAccessKey -WithDecryption $TRUE).Value
+$sSecretKey=(Get-SSMParameter -Name SSecretKey -WithDecryption $TRUE).Value
+$accessKey=(Get-SSMParameter -Name AccessKey -WithDecryption $TRUE).Value
```

Comment:
what do you think about relying on the instance role?

Reply:
I mean, relying on the instance role for standard partition. These are the keys for both partitions, right? If this is more straightforward, lets keep it.